### PR TITLE
Add dollarsToCentsSafe function

### PR DESCRIPTION
**Pull request by wwahammy**
- Add a dollarsToCentsSafe function
- Add better return types
- Add some documentation to dollarsToCentsSafe

**NOTE: DO NOT discuss internal CommitChange information in your PR; this PR will be public.
Link back to the issue in the Tix repo when you need to do that.**

### CODE REVIEW

File: javascripts/src/lib/format.spec.ts [hunk 1]
```diff
@@ -19,4 +19,45 @@ describe('Format.dollarsToCents', () => {
...
+
+  test("properly throws on undefined",() =>
+    expect(() => { Format.dollarsToCents(undefined)}).toThrowError()
+  )
+
+})
+
+describe('Format.dollarsToCentsSafe', () => {
+
+  const expectedAmount = 120
```
Comment (camckin10) [inline javascripts/src/lib/format.spec.ts:38]: Was this expectedAmount created only for testing? 
Or, is this a number that CommitChanges uses regularly in the code?

File: javascripts/src/lib/format.ts [hunk 2]
```diff
@@ -15,7 +15,13 @@ export function centsToDollars(cents:string|number|undefined, options:{noCents?:
   return numberWithCommas((centsAsNumber / 100.0).toFixed(options.noCents ? 0 : 2).toString()).replace(/\.00$/,'')
 }
 
-export function dollarsToCents(dollars:string) {
+/**
+ * Convert a string describing a dollar into a number representing the cents. 
+ * @param dollars a string representing an amount in dollars
```
Comment (camckin10) [inline javascripts/src/lib/format.ts:20]: Syntax question:

When you add an '@' symbol in front of these words, does that let the library know that: 
a) these lines of code serve like comment lines? 
b) these lines of code should not be run

File: javascripts/src/lib/format.spec.ts (hunk 1 above)
Comment (wwahammy) [reply javascripts/src/lib/format.spec.ts:38]: I don't remember why I picked that number in the original for dollarsToCents test. My hunch is that it was because it was mostly a random choice. That said, I suspect part of it was that it has a number with a trailing zero. Since dollar strings written as "1.20" and "1.2" would both be converted to 120, that would help test that corner case a bit more.

File: javascripts/src/lib/format.ts (hunk 2 above)
Comment (wwahammy) [reply javascripts/src/lib/format.ts:20]: Everything from lines 18-23 is a multi-line comment. The tags like `@param` , `@returns`, etc. are [JSDoc tags](https://jsdoc.app/). They're a standardized way to document functions (and classes) in a Javascript, or in this case Typescript, file. They're machine readable and a lot of tools use them to add richer info to documentation. For example, VSCode would provide additional documentation if you're using the function in another file.

File: javascripts/src/lib/format.spec.ts (hunk 1 above)
Comment (camckin10) [reply javascripts/src/lib/format.spec.ts:38]: Got it!

File: javascripts/src/lib/format.ts (hunk 2 above)
Comment (camckin10) [reply javascripts/src/lib/format.ts:20]: 😮  Nice! Thank you!

**Review by camckin10 — APPROVED** [review]
lgtm.
